lib/oli_web/live/delivery/student/content_live.ex

```diff
@@ -3,23 +3,391 @@ defmodule OliWeb.Delivery.Student.ContentLive do
+            "flex flex-col gap-[5px] cursor-pointer rounded-xl h-[162px] w-[288px] shrink-0 mb-1 px-5 pt-[15px] bg-gray-200",
+            if(@selected, do: "bg-gray-400 outline outline-2 outline-gray-800"),
+            if(@bg_image_url in ["", nil],
+              do: "bg-[url('/images/course_default.jpg')]",
+              else: "bg-[url('#{@bg_image_url}')]"
+            )
+          ]}
+        >
+          <span class="text-[12px] leading-[16px] font-bold opacity-60 text-gray-500">
+            <%= "#{@unit_numbering_index}.#{@module_index}" %>
+          </span>
+          <h5 class="text-[18px] leading-[25px] font-bold"><%= @module.revision.title %></h5>
+          <div :if={!@selected} class="mt-auto flex h-[21px] justify-center items-center">
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="10"
+              height="5"
+              viewBox="0 0 10 5"
+              fill="currentColor"
+            >
+              <path d="M0 0L10 0L5 5Z" />
+            </svg>
+          </div>
+        </div>
+        <.progress_bar :if={!@selected} percent={:rand.uniform(100)} width="60%" show_percent={false} />
+        <div
+          :if={@selected}
+          class={[
+            "flex justify-center items-center -mt-1"
+          ]}
+        >
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="27"
+            height="12"
+            viewBox="0 0 27 12"
+            fill="currentColor"
+          >
+            <path d="M0 0L27 0L13.5 12Z" />
+          </svg>
+        </div>
+      </div>
+    </div>
+    """
+  end
+
+  attr(:percent, :integer, required: true)
+  attr(:width, :string, default: "100%")
+  attr(:show_percent, :boolean, default: true)
+
+  def progress_bar(assigns) do
+    ~H"""
+    <div class="flex flex-row items-center mx-auto">
+      <div class="flex justify-center w-full">
+        <div class="rounded-full bg-gray-200 h-1" style={"width: #{@width}"}>
+          <div class="rounded-full bg-[#1E9531] h-1" style={"width: #{@percent}%"}></div>
+        </div>
+      </div>
+      <div :if={@show_percent} class="text-[16px] leading-[32px] tracking-[0.02px] font-bold">
+        <%= @percent %>%
+      </div>
+    </div>
+    """
+  end
+
+  _docp = """
+    Currently the intro_content for a revision does not support h1 tags. So,
+    if there is no <h1> tag in the content then we need to add an aditional margin
+    to match the given figma design.
+  """
+
+  defp maybe_additional_margin_top(content) do
+    if !String.contains?(Jason.encode!(content), "\"type\":\"h1\""), do: "mt-[52px]"
+  end
+
+  defp parse_datetime(nil, _ctx), do: "not yet scheduled"
+
+  defp parse_datetime(datetime, ctx) do
+    datetime
+    |> FormatDateTime.convert_datetime(ctx)
+    |> Timex.format!("{WDshort} {Mshort} {D}, {YYYY}")
+  end
+
+  defp week_number(_section_start_date, nil), do: 1
+
+  defp week_number(section_start_datetime, unit_start_datetime) do
```

Comment:
Yes! I moved it to `OliWeb.Components.Delivery.Utils` and made a small refactor to use `trunc()` to get the integer part of the float (instead of doing: `float_number |> Float.to_string() |> Integer.parse()`)